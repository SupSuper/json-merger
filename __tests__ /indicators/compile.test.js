const helpers = require("../../__helpers__");

describe("when merging two objects and a source property has a $compile indicator", function () {

    test("it should compile the $compile property and then compile/merge the result with the target", function () {

        const object1 = {
            "a": {
                "aa": "original"
            }
        };

        const object2 = {
            "a": {
                "$compile": {
                    "$set": {
                        "key": "$value",
                        "value": "replaced"
                    }
                }
            }
        };

        const result = helpers.mergeObjects([object1, object2]);

        expect(result).toMatchSnapshot();
    });

    test("it should use the $compile property result as $sourceRoot when compiling/merging with the target", function () {

        const object = {
            "$compile": {
                "$merge": {
                    "source": {
                        "a": {
                            "$set": {
                                "key": "$expression",
                                "value": "$sourceRoot.b"
                            }
                        }
                    },
                    "with": {
                        "b": "bValue"
                    }
                }
            }
        };

        const result = helpers.fromObject(object);

        expect(result).toMatchSnapshot();
    });
});
