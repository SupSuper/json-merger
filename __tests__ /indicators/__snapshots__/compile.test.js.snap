// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when merging two objects and a source property has a $compile indicator it should compile the $compile property and then compile/merge the result with the target 1`] = `
"{
	\\"a\\": \\"replaced\\"
}"
`;

exports[`when merging two objects and a source property has a $compile indicator it should use the $compile property result as $sourceRoot when compiling/merging with the target 1`] = `
"{
	\\"a\\": \\"bValue\\",
	\\"b\\": \\"bValue\\"
}"
`;
